usr/sbin/setup-fstab-for-overlayfs

```diff
@@ -1,32 +1,22 @@
-#!/bin/sh
+#!/bin/sh -e
 #
 # Copyright (c) 2019 SUSE Linux GmbH, Nuernberg, Germany
 #
 # Add mountpoint for the /etc overlay and workaround for /var to /etc/fstab
 #
 
 # Already there?
-if [ -e /etc/fstab ] && grep -qE '^overlay[[:space:]]+/etc[[:space:]]' /etc/fstab; then
+if [ -e /etc/fstab ] && grep -qE '^/etc[[:space:]]+/etc[[:space:]]+none[[:space:]]+bind' /etc/fstab; then
 	exit 0 # Do nothing
 fi
 
-# Not a fresh installation?
-if [ -e /etc/fstab.sys ]; then
-	# Let transactional-update handle the migration - doing anything here would be pointless
-	# as it's only written into the overlay anyway and if for some reason transactional-update
-	# is too old, it would result in duplicate /etc entries.
-	exit 0
-fi
+# Migration from old system
+echo "Migrating from /etc overlays to /etc subvolume..."
+btrfs subvolume create /etc
```

Comment:
Maybe in the future `<volume name="etc"/>` is added in kiwi and in the skelcd, requiring first a check here?

Reply:
From what I know this (currently) won't work because KIWI doesn't support creating nested BTRFS subvolumes.

But in any case I've changed the check to generally exit if `/etc` appears in the _fstab_ file now. Additionally it will explicitly `exit 1` now - this script is only supposed to be run during installation, so we will see if something changes somewhere else in the setup, and an existing _/etc_ would definitely conflict with this script.